keps/sig-api-machinery/2161-apiserver-default-labels/README.md

```diff
@@ -0,0 +1,487 @@
+|                                            | |  |
+|                                            | |  |
+|                                            | |  |
++--------------------------------------------+ |  | <-----------+ defaults.go mutates
+                                               +--+               namespaces on read
+                                                                  to always have
+                                                                  the label
+                                                                  kubernetes.io/metadata.name=obj.Name
+```
+
+### User Stories (Optional)
+
+#### Avoid labels misuse
+
+NetworkPolicies allow users to open up traffic to/from their namespaces with the help of `namespaceSelector` field.
+However, there is a concern that arbitrary labels may not be the most secure way to open up traffic when it comes
+to namespace resource. In case a namespace's labels are known by others, any user can add labels at any time to any
+namespace and thus send traffic to a pod belonging to this namespace. Matching a namespace by its name, on the other
+hand, is a more reliable way to whitelist namespaces as it's much easier to specifically allow only namespaces user
+control since user knows their own namespace names.
+
+#### Lack of permissions
+
+Users want to allow DNS traffic to coreDNS pods which are created in system generated namespaces like kube-system.
+Since these system-generated namespaces do not come up with labels, users may not be authorized to add labels to
+such namespaces and thereby accessing some essential services is not straight forward.
+
+#### Simplicity
+
+It is being reported that users find it difficult to work with mutation and validation webhooks as they are not
+able to select a single namespace to enforce those controllers. Similarly, it is quite common and intuitive for
+users to remember the names of the namespaces rather than labels associated with those namespaces while defining
+NetworkPolicies or writing admission controller configurations.
+
+
+### Notes/Constraints/Caveats (Optional)
+
+* The name of the namespace must satisfy label value [requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set).
+Currently, the name of the namespace must be a valid [DNS label](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names),
+and this requirement must be satisfied going forward.
+
+### Risks and Mitigations
+
+* Every namespace gets bigger in size. We think this is inconsequential
+* Some user may already be using that label already. This is mitigated by the 
+fact that the whole "kubernetes.io" label prefix is [reserved](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set). However, since there
+is no validation of reserved label prefixes, an upgrade would mutate such a label.
```

Comment:
given
* how long the label key is
* and its english meaning

I consider it extremely unlikely that someone would be using this label already *and* not desire that it have the value we want it to have.

Reply:
yeah, agree.

Reply:
pushed , thx